### PR TITLE
LC-263 - Display audiobooks on collection items.

### PR DESCRIPTION
**Pull request by bozeak**
- Added `audiobooks` support
- Personalized and translatable buttons labels ("Preview ebook", "Preview audiobook", "Preview videos")
- Other fixes

### CODE REVIEW

File: easyopac_collection_links.module [hunk 1]
```diff
@@ -130,23 +136,63 @@ function easyopac_collection_links_ding_entity_buttons($type, $entity) {
...
+ *   Requested item material type.
  *
  * @return mixed
  */
-function easyopac_collection_links_generate_preview_url(array $preview) {
+function easyopac_collection_links_generate_preview_url(array $preview, $type) {
+  $mtype = '';
   $preview_url = NULL;
+  $types = [
+    'ebook' => [
```
Comment (guddo) [inline easyopac_collection_links.module]: This does not look right

**Review by guddo — CHANGES_REQUESTED** [review]
I'm sure  the types are wrong

File: easyopac_collection_links.module (hunk 1 above)
Comment (bozeak) [reply easyopac_collection_links.module]: Maybe it looks wrong, but during debugging, development and testing I got responses from LMS where external resources was belonging to those types.

Comment (guddo) [reply easyopac_collection_links.module]: What? What does these types define?

Comment (bozeak) [reply easyopac_collection_links.module]: In this array is defined mapping of material type (Bog, Ebog, Lydbog etc.) to the types of external resources (ebook, audiobook, videos).

Comment (guddo) [reply easyopac_collection_links.module]: You mean on which types in LMS they are shown, or where they are available?

This is the match in LMS:
- ebook = face.type="ebog" AND facet.acSource="ereolen"
- audiobook = facet.type="lydbog (net)" AND facet.acSource="netlydbog"

Comment (bozeak) [reply easyopac_collection_links.module]: Which types in LMS they are shown. I'm working with `/details` response object (Ex: https://stg.lms.inlead.ws/hedbib/detail?faustNumber=47640962&withExternalResources) and the info I can use for correct mapping of a external resource link to EO material item can be taken only from `type` and `externalResources` keys.

Comment (guddo) [reply easyopac_collection_links.module]: Sorry I'm not sure I understand. All item types in LMS can possibly contain these 2 values.

**Review by guddo — APPROVED** [review]

**Pull request merged**